src/lib/libast/path/pathprog.c

```diff
@@ -35,8 +35,7 @@
 #include <ctype.h>
 #endif
 
-#include "FEATURE/prog"
-
+/* TODO: Is it needed ? */
```

Comment:
If this isn't needed then neither is the /proc/self stuff above.  So you'll want to add

```
if compiler.has_header('mach-o/dyld.h')
    libast_c_args += [ '-D_hdr_mach_o_dyld']
endif
```

It should be possible to omit the check for the `_NSGetExecutablePath` function since if the header exists we should be able to assume we're building on macOS and the function exists.